src/objective-c/GRPCClient/private/NSData+GRPC.m

```diff
@@ -42,7 +42,11 @@
 static void MallocAndCopyByteBufferToCharArray(grpc_byte_buffer *buffer,
                                                size_t *length, char **array) {
   grpc_byte_buffer_reader reader;
-  grpc_byte_buffer_reader_init(&reader, buffer);
+  if (!grpc_byte_buffer_reader_init(&reader, buffer)) {
```

Comment:
That seems to expose internal details that may not be true in the future. All this code should care about is that the initialization of the reader failed. Arguably, we may want to add some function (eg, `grpc_byte_buffer_reader_status`) expanding on the reason for the failure. How does that sound?